e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -507,9 +601,9 @@ const openDashboard = (params = {}) => {
 };
 
 const openSlowDashboard = (params = {}) => {
-  cy.intercept("POST", "/api/dashboard/*/dashcard/*/card/*/query", req => {
-    return Cypress.Promise.delay().then(() => req.reply());
-  }).as("cardQuery");
+  cy.intercept("POST", "/api/dashboard/*/dashcard/*/card/*/query").as(
```

Comment:
From testing, we don't need the last part that tries to delay the response anymore. The current tests are almost mimicking the behavior of slow API responses, but not fully. It works by calling `cy.tick(MORE_THAN_15_SECONDS)` before the API is returned so when the dashboard slow timeout (15s) is called and marks a dashboard as slow.

What I wish we could do is to mock the API to **actually** return after 15 seconds. However, I'm not sure how to do that when working with `cy.clock()`

we could use `delay` option in `cy.intercept`, that actually delays the response by the specified milliseconds, but it doesn't work with `cy.clock()` or `cy.tick()` so we need to actually **wait** for the amount of delay in each test. That's why I think this compromise is good enough to simulate the slow dashboard behavior.